lib/new_relic/agent/transaction/tracing.rb

```diff
@@ -29,6 +29,12 @@ def add_segment(segment, parent = nil)
           else
             segment.record_on_finish = true
             ::NewRelic::Agent.logger.debug("Segment limit of #{segment_limit} reached, ceasing collection.")
+
+            if finished?
+              ::NewRelic::Agent.logger.debug("Transaction #{best_name} has finished but segments still being created, resetting state.")
```

Comment:
This new module code make use of a method, `best_name`, that doesn't exist in the module, so the code can only be used by an instance of the `Transaction` class. Seeing as the `Transaction` class is the only consumer (includer) of this `Tracing` module, this is probably ok.